### ISSUE TITLE
Dual value of constraints specified as variable attributes.

### ISSUE THREAD

**Issue by marchinidavide** [cvxpy/cvxpy#1863]
More of a question than a feature request.
Currently calling `problem.constraints()` does not include the "implicit" constraints defined as variables attributes. 
From the docs it looks like specifying variable constraints through attributes is the recommended practice.

```python
# Non negative least squares
import cvxpy as cp
import numpy as np

rng = np.random.default_rng(seed=0)
A = rng.standard_normal((20, 4))
b = rng.standard_normal(20)
x = cp.Variable(shape=(4, ), nonneg=True)
obj =  cp.Minimize(cp.sum_squares(b - A @ x))
problem = cp.Problem(objective=obj, constraints=[])
_ = problem.solve()
print(problem.constraints)  # empty list
```

I found that `problem.variables()[0].domain[0].dual_value` of the solved problem is None.
However, specifying `constraints=[x >= 0]` and then calling `problem.constraints[0].dual_value` returns `array([0.        , 1.66159569, 0.        , 0.        ])`.

Any hint on what I'm doing wrong?

**Comment by SteveDiamond** [comment]
Hmm that's a good point, you currently can't access dual values for variable attributes. We can add documentation saying if you want dual variables you should specify the constraint explicitly.

**Comment by SteveDiamond** [comment]
Yes that hypothetically could happen but could be fixed using a wrap (https://www.cvxpy.org/_modules/cvxpy/atoms/affine/wraps.html).

**Comment by rileyjmurray** [comment]
@SteveDiamond would you be open to adding functionality where dual variables of attributes are propagated back from the solver?

**Comment by SteveDiamond** [comment]
I'm not sure this feature should be added. Wouldn't we need dual values for attributes that don't generate constraints like symmetric?

**Comment by rileyjmurray** [comment]
That's a good point. As far as symmetric / Hermitian goes, it would always be valid to set a dual variable to the matrix of all zeros (and to do that lazily).

**Comment by SteveDiamond** [comment]
Thinking about it more, I'm ok with adding dual variables for the attributes nonneg, nonpos, PSD, and NSD, accessible via ``domain`` as above. Covering other attributes is not so essential, and they don't have ``domain`` so the access pattern above wouldn't even be possible.
